src/state/slices/actionSlice/selectors.ts

```diff
@@ -117,14 +137,18 @@ export const selectOpenLimitOrderActionsFilteredByWallet = createDeepEqualOutput
   },
 )
 
-export const selectLimitOrderActionByCowSwapQuoteId = createDeepEqualOutputSelector(
-  selectActions,
+export const selectWalletLimitOrderActionByCowSwapQuoteId = createDeepEqualOutputSelector(
+  selectWalletActions,
   selectCowSwapQuoteIdParamFromRequiredFilter,
   (actions, cowSwapQuoteId) => {
-    return actions.find(
+    const maybeAction = actions.find(
       action =>
         isLimitOrderAction(action) && action.limitOrderMetadata.cowSwapQuoteId === cowSwapQuoteId,
     )
+
+    if (!maybeAction || !isLimitOrderAction(maybeAction)) return
```

Comment:
type narrowing from the predicate lost - would be kept if we were simply doing `actions.find(action =>isLimitOrderAction(action))` but the second call voids narrowing